src/hooks/useConnectorImagesSpec.ts

```diff
@@ -1,57 +1,61 @@
+import {
+    ConnectorImagesSpecResponse,
+    connectorsEndpoint,
+} from 'endpoints/connectors';
+import { useAsync } from 'hooks/useAsync';
 import JsonRefs from 'json-refs';
-import { useEffect, useState } from 'react';
-import axios, { withAxios } from 'services/axios';
-import { type BaseHook } from 'types';
-
-interface ConnectorImagesService extends BaseHook {
-    data: {
-        specSchema: any;
-        links: Pick<NewCaptureState['links'], 'discovery' | 'documentation'>;
+import { useEffect } from 'react';
+
+interface Data {
+    endpointSchema: ConnectorImagesSpecResponse['data']['attributes']['endpointSpecSchema'];
+    links: {
+        discovery: ConnectorImagesSpecResponse['data']['links']['discovery'];
+        documentation: ConnectorImagesSpecResponse['data']['attributes']['documentationURL'];
     };
 }
 
-const useConnectorImageSpec = (specURL: string): ConnectorImagesService => {
-    const [schema, setSchema] = useState<object>({});
-    const [error, setError] = useState<string | null>(null);
-    const [discovery, setDiscovery] = useState<string>('');
-    const [docs, setDocs] = useState<string>('');
-    const [loading, setLoading] = useState<boolean>(true);
+const defaultData: Data = {
+    endpointSchema: {},
+    links: {
+        discovery: '',
+        documentation: '',
+    },
+};
 
-    useEffect(() => {
-        if (specURL) {
-            setLoading(true);
-            setError(null);
-            withAxios(axios.get(specURL), setError, setLoading)
-                .then((specResponse: any) => {
-                    const { data } = specResponse.data;
-                    JsonRefs.resolveRefs(data.attributes.endpointSpecSchema)
-                        .then((derefSchema) => {
-                            setSchema(derefSchema.resolved);
-                        })
-                        .catch((resolveRefError) => {
-                            setSchema({});
-                            setError(resolveRefError.message);
-                        });
-
-                    setDiscovery(data.links.discovery);
-                    setDocs(data.attributes.documentationURL);
-                })
-                .catch(() => {});
-        }
-    }, [specURL]);
+const generateResponse = async (
+    endpointResponse: ConnectorImagesSpecResponse
+) => {
+    const { data } = endpointResponse;
+    const derefSchema = await JsonRefs.resolveRefs(
+        data.attributes.endpointSpecSchema
+    );
 
-    return {
-        data: {
+    return new Promise<Data>((resolve) => {
+        resolve({
+            endpointSchema: derefSchema.resolved,
             links: {
-                discovery,
-                documentation: docs,
+                discovery: data.links.discovery,
+                documentation: data.attributes.documentationURL,
             },
-            specSchema: schema,
-        },
-        error,
-        loading,
-    };
+        });
+    });
+};
```

Comment:
I think keeping the "post processing" handled in a stand along function makes things cleaner down below. Also... should end up using this pattern by exporting this and having folks that need it run run it themselves vs always running / passing switches.